macros/src/symbol_process.rs

```diff
@@ -570,7 +570,17 @@ impl DollarRefsScope {
         let upstream = self
           .iter()
           .map(|DollarRef { name, .. }| quote! {  #name.modifies() });
-        quote! { observable::from_iter([#(#upstream),*]).merge_all(usize::MAX) }
+        let readers = self
+          .iter()
+          .map(|DollarRef { name, .. }| quote! {  let #name = #name.clone_reader(); });
+        let checks = self
+          .iter()
+          .map(|DollarRef { name, .. }| quote! {  #name.is_valid() });
+        quote! { observable::from_iter([#(#upstream),*]).merge_all(usize::MAX).filter({
```

Comment:
instead of use `filter`, `take_until` can unsubscribe when the reader is invalid.